### ISSUE TITLE
yt-dlp 2023.06.21 doesn't download YT Premium Video Streams by default

### ISSUE THREAD

**Issue by ThePhoenix576** [yt-dlp/yt-dlp#7371]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.06.21** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

Hello, So on previous releases like for example yt-dlp 2023.03.04, when yt-dlp is provided with a YT link with a premium enhanced bitrate video stream. It will select that over the lower bitrate video stream. But now in this new version it prefers https regardless even though there is a premium bitrate video stream available. 

For context here is the link of a video that has premium 1080p video stream = https://www.youtube.com/watch?v=3QgLwpuDGhI

And also for these premium bitrate videos, you don't need to pass cookies with a real premium account for it to work

And here is the mediainfo bitrate details comparing 248 and 616 video streams (No Audio, just plain 248 and 616 stream) : 
<img width="433" alt="image" src="https://github.com/yt-dlp/yt-dlp/assets/40349017/a0a6b895-8472-4d0c-bdbe-ab498e587b7c">

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--extractor-args', 'youtube:player_client=default,ios', 'https://www.youtube.com/watch?v=3QgLwpuDGhI']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.06.21 [42f2d40b4] (pip)
[debug] Python 3.10.11 (CPython arm64 64bit) - macOS-13.4-arm64-arm-64bit (OpenSSL 3.1.1 30 May 2023)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2023.05.07, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1848 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.06.21, Current version: stable@2023.06.21
yt-dlp is up to date (stable@2023.06.21)
[youtube] Extracting URL: https://www.youtube.com/watch?v=3QgLwpuDGhI
[youtube] 3QgLwpuDGhI: Downloading webpage
[youtube] 3QgLwpuDGhI: Downloading android player API JSON
[youtube] 3QgLwpuDGhI: Downloading ios player API JSON
[youtube] 3QgLwpuDGhI: Downloading m3u8 information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 3QgLwpuDGhI: Downloading 1 format(s): 248+251
[debug] Invoking http downloader on "https://rr5---sn-poqvn5u-jb36.googlevideo.com/videoplayback?expire=1687355922&ei=sq2SZKH_J9ea1d8PzKWrmAM&ip=2407%3A0%3A3002%3A8bb%3A95bb%3Abccc%3A17b0%3Aaa72&id=o-AIFQ_bM6cuCVWM4Tx0cBoIir7kw8RJ0P9mlHAfmZ4fES&itag=248&source=youtube&requiressl=yes&mh=wh&mm=31%2C26&mn=sn-poqvn5u-jb36%2Csn-30a7rned&ms=au%2Conr&mv=m&mvi=5&pl=52&initcwndbps=297500&spc=qEK7B39ZtjYAUpsAu-Shk1HfH4q0Vi0&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=213429249&dur=1636.900&lmt=1686947948333681&mt=1687333978&fvip=2&keepalive=yes&fexp=24007246%2C24363392&beids=24350017&c=ANDROID&txp=5535434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgMzyqRlYVRIoqbgpKr23wcas-o1uaCMtu1YwJ8IjbuDgCIQCmquyTfkeFfU4VE4KICI-TvJrs4S1Y95MicFNnWf1WPg%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhAOdLruSBm3_184ZENpYINIVZU7V_psH7IGcTH9GNmQNqAiEAvtW9uuZ_XkAwE1sB2v8a8-rR7AtghLsMWmxBKhXXU3s%3D"
[download] Destination: New study claims Betelgeuse supernova IMMINENT (decades not centuries!) ｜ Night Sky News June 2023 [3QgLwpuDGhI].f248.webm
[download]  20.0% of  203.54MiB at   10.92MiB/s ETA 00:14^C
ERROR: Interrupted by user
```

**Comment by pukkandan** [comment]
In the previou release, it was getting sorted below by accident (because we were not extracting https formats). We haven't yet figured out how to properly sort the `ios` formats since youtube doesn't give any indicator it's premium format. You can use something like `-S res,vcodec:vp9,tbr` in the meantime.

**Comment by ThePhoenix576** [comment]
> In the previous release, it was getting sorted below by accident (because we were not extracting https formats). We haven't yet figured out how to properly sort the `ios` formats since youtube doesn't give any indicator it's premium format. You can use something like `-S res,vcodec:vp9,tbr` in the meantime.

Yeah, it is weird how they have only partly released it to the public. And yeah it is weird how they didn't give any indicator that it is a premium bitrate version. Anyhow, I am not that familiar with -S argument. Does it work fine when used with -f custom formats? Thank you

**Comment by pukkandan** [comment]
Yes, they work together

**Comment by bashonly** [comment]
Fixed in 1e75d97db21152acc764b30a688e516f04b8a142

Will be fixed in next release. If you don't want to wait, you can [install from master with pip](https://github.com/yt-dlp/yt-dlp/wiki/Installation#pip-master) (or run `yt-dlp --update-to nightly` with a binary/executable)
